File: golang/_0009_Palindrome_Number/main.go
package main

func isPalindrome(x int) bool {
	if x == 0 {
		return true
	} else if x < 0 || x%10 == 0 {
		return false
	}

	rx := 0
	for x > rx {
		if x/10 == rx {
			break
		}
		rx = rx*10 + x%10
		x /= 10
	}

	return x == rx || x/10 == rx
}

File: python3/_0009_Palindrome_Number/main.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True
        elif x < 0 or x % 10 == 0:
            return False

        rx = 0
        while x > rx:
            if rx == x // 10:
                break

            rx = rx * 10 + x % 10
            x //= 10

        return x == rx or rx == (x // 10)
